python/index_lambda.py

```diff
@@ -0,0 +1,129 @@
+list_of_contacts = ['+14049604358','+14704944634']
+COLLECTION = str(os.environ.get('COLLECTION'))
+CROPBUFFER = int(os.environ.get('BUFFER'))
+
+def findFaces(collection, srcBucket, srcKey):
+    client = boto3.client('rekognition')
+    #Find faces in the image
+    try:
+        response = client.detect_faces(
+            Image={
+                'S3Object': {
+                    'Bucket': srcBucket,
+                    'Name': srcKey
+                }
+            },
+            )
+    except Exception as e:
+        print "Your exception is: ",e
+        return 'Code exception in findfaces'
+
+    print 'Number of faces:',len(response['FaceDetails'])
+    if len(response['FaceDetails'])!=0:
+        return crop(response,srcBucket,srcKey)
+    else:
+        return 'No faces found.'
+
+
+def crop(data, srcBucket, srcKey):
+    client = boto3.client('s3')
+    s3 = boto3.resource('s3')
+    localFilename = '/tmp/{}'.format(os.path.basename(srcKey))
+
+    #Download image to local
+    try:
+        s3.Bucket(srcBucket).download_file(srcKey, localFilename)
+    except botocore.exceptions.ClientError as e:
+        if e.response['Error']['Code'] == "404":
+            print("The object does not exist.")
+        else:
+            raise
+
+    # Load the original image:
+    img = Image.open(localFilename)
+    for i in range (0,len(data['FaceDetails'])):
+        width = data['FaceDetails'][i]['BoundingBox']['Width']
+        height = data['FaceDetails'][i]['BoundingBox']['Height']
+        left = data['FaceDetails'][i]['BoundingBox']['Left']
+        top =  data['FaceDetails'][i]['BoundingBox']['Top']
+
+        widImage = img.size[0]
+        htImage = img.size[1]
+
+        #Added Buffer to avoid tight cropping and provide room for detecting faces
+
+        img1 = img.crop(((left * widImage) - CROPBUFFER,(top * htImage) - CROPBUFFER,(left * widImage) + (width * widImage) + (2 * CROPBUFFER),(top * htImage) + (height * htImage) + (2 * CROPBUFFER)))
+
+        key = srcKey.replace('/',"")
+        #naming the image
+        tmpCropped = 'cropped_'+str(i)+key
+        
+        #Changing image to bytes
+        imgByteArr = io.BytesIO()
+        img1.save(imgByteArr, format='JPEG')
+        imgByteArr = imgByteArr.getvalue()
+        response = searchImageinCollection(COLLECTION, srcBucket, imgByteArr)
+
+        #If no matches found add the image to test folder
+        if response != False and len(response['FaceMatches'])==0:
+            print 'Alert'
+            name = 'test/'+tmpCropped
+            #put object in bucket
+            object = s3.Bucket(srcBucket).put_object(Body = imgByteArr, Key = name)
+
+            sns = boto3.client("sns")
+
+            #image url
+            url = client.generate_presigned_url(ClientMethod='get_object',Params={'Bucket': srcBucket,'Key': name})
+
+            # Send your sms message.
+            for number in list_of_contacts:
+                sns.publish(PhoneNumber = number, Message=url)
```

Comment:
when calling publish we should  be publishing to a TopicArn not a phonenumber. This way we can build on the notification and do whatever we want with it. http://boto3.readthedocs.io/en/latest/reference/services/sns.html#SNS.Client.publish

You will need to create a topic arn in aws. If you need help with this let me know.